src/test/scala/com/eclipsesource/schema/SchemaValidatorSpec.scala

```diff
@@ -165,39 +167,37 @@ class SchemaValidatorSpec extends PlaySpecification {
       }
     }
 
-  "should resolve references on the classpath via UrlResolver" in {
-    val validator = SchemaValidator().addUrlResolver(ClasspathUrlResolver())
+  "should resolve references on the classpath with ClasspathUrlResolver" in {
+    val validator = SchemaValidator().addUrlResolver(ClasspathUrlProtocolHandler())
     // some.json references location.json within JAR
     val someJson = getClass.getResourceAsStream("/some.json")
     val schema = JsonSource.schemaFromStream(someJson)
     val result = validator.validate(schema.get, Json.obj("location" -> Json.obj("name" -> "Munich")))
     result.isSuccess must beTrue
   }
 
-  "should resolve references on the classpath via UrlResolver and custom protocols (#65)" in {
+  "should resolve relative references on the classpath with relative URL handler (#65)" in {
     val validator = SchemaValidator()
-      .addUrlResolver(ClasspathUrlResolver())
-      .shouldResolveRelativeRefsWithCustomProtocols(true)
-    // some.json references location.json within JAR
+      .addRelativeUrlHandler(new ClasspathHandler)
+    // some-issue-65.json references location.json within JAR
     val someJson = getClass.getResourceAsStream("/some-issue-65.json")
     val schema = JsonSource.schemaFromStream(someJson)
     val result = validator.validate(schema.get, Json.obj("location" -> Json.obj("name" -> "Munich")))
     result.isSuccess must beTrue
   }
 
-  "should resolve relative references with custom protocols (#65)" in {
+  "should resolve relative references on the classpath with relative URL handler (#65 - 2)" in {
     val validator = SchemaValidator()
-      .addUrlResolver(ClasspathUrlResolver())
-      .shouldResolveRelativeRefsWithCustomProtocols(true)
+      .addRelativeUrlHandler(new ClasspathHandler)
     val mySchemaJson = getClass.getResourceAsStream("/my-schema.schema")
     val mySchema = JsonSource.schemaFromStream(mySchemaJson)
     val result = validator.validate(mySchema.get, Json.obj("foo" -> Json.obj("bar" -> "whatever")))
     result.isSuccess must beTrue
   }
 
-  "should fail resolving relative references with custom protocols if option not set(#65)" in {
+  "should fail resolving relative references without any relative URL handler available (#65)" in {
     val validator = SchemaValidator()
-      .addUrlResolver(ClasspathUrlResolver())
+      .addUrlResolver(ClasspathUrlProtocolHandler())
```

Comment:
@andrewmeyerBR I've found some time the last couple of days to implement your proposed suggestions as well as a couple of test cases [here](https://github.com/eclipsesource/play-json-schema-validator/blob/master/src/test/scala/com/eclipsesource/schema/UrlHandlerSpec.scala). For relative refs it is crucial to load the schema via an URL, otherwise the resolution scope can't be inferred. Also, a custom handler for relative refs to resources on the classpath shouldn’t be necessary at all any more, since when using the absolute path during URL construction (as discussed, the absolute URL is now passed to the relative URL handlers) the already available default URLStreamHandlers (e.g. the one handling the `jar` scheme) are used. You should be able to override that behaviour, if necessary, by registering an URL handler for the respective schema. 
The test cases are not as exhaustive as they could be currently, but they do feature a `MyUrlHandler` example URL handler that demonstrates that protocol-less relative refs resolution is possible. In the example, there's a  protocol-less relative ref pointer to `quux.schema`, which doesn't actually exist. The `MyUrlHandler` simply returns the existing `foo.schema` in that case.
If there any questions or bugs, please open an issue in order to continue our discussion there instead of using this PR. Thank you!